train_diffusion_prior.py

```diff
@@ -252,20 +253,31 @@ def main():
     args = parser.parse_args()
 
     print("Setting up wandb logging... Please wait...")
-
-    wandb.init(
-      entity=args.wandb_entity,
-      project=args.wandb_project,
-      config={
-      "learning_rate": args.learning_rate,
+    
+    config = ({"learning_rate": args.learning_rate,
       "architecture": args.wandb_arch,
       "dataset": args.wandb_dataset,
       "Learning Rate":args.learning_rate,
       "Weight Decay":args.weight_decay,
       "Max Gradient Clipping Norm":args.max_grad_norm,
       "Batch size":args.batch_size,
-      "epochs": args.num_epochs
-      })
+      "epochs": args.num_epochs,
```

Comment:
can you use these params in the resume part ?

Reply:
Ok, fixed - the hyperparameters of the pretrained model are now being used - only in case of RESUME.

Reply:
No you didn't understand what I mean.

In resume you are not using the hyperparameters from the checkpoint
You are still using the parameters from argparse
https://github.com/lucidrains/DALLE2-pytorch/blob/6ef5872a35ff01c55da80c6cf9b3c4ee5ffb31c0/train_diffusion_prior.py#L107 there

When doing that, you will notice it would be much more convenient to have a subdict for each network with exactly the same names as the constructor of the network so let's do that as well

Reply:
Fixed, do take a look .

Reply:
Quick point - having the same names as the constructor might not work - to do a **dpn_config , because the DPN constructor uses a parameter from the dp_config(i.e  dp_normformer).